guava/src/com/google/common/io/ByteStreams.java

```diff
@@ -816,7 +816,8 @@ public static void skipFully(InputStream in, long n) throws IOException {
    */
   static long skipUpTo(InputStream in, final long n) throws IOException {
     long totalSkipped = 0;
-    byte[] buf = createBuffer();
+    // A buffer is allocated if skipSafely does not skip any bytes.
+    byte[] buf = null;
```

Comment:
It may be worth considering a singleton skip-buffer instead of allocation in this case. Using that model, it's possible to write a stream to snoop on other skip calls sharing the JVM which may be reading sensitive data, so I opted for an approach more similar to the original implementation.

Reply:
Yes, that's exactly why we aren't using a singleton buffer for this.